pkg/component/etcd/statefulset/statefulset.go

```diff
@@ -0,0 +1,239 @@
+		return err
+	}*/
+
+	return nil
+}
+
+func (c *component) Destroy(ctx context.Context) error {
+	etcdMainSts := c.emptyStatefulset(c.values.EtcdMainStsName)
+
+	if err := c.deleteSts(ctx, etcdMainSts); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *component) syncEtcdMainSts(ctx context.Context, sts *appsv1.StatefulSet) error {
+	_, err := controllerutils.GetAndCreateOrStrategicMergePatch(ctx, c.client, sts, func() error {
+		sts.ObjectMeta = getObjectMeta(&c.values)
+		sts.Spec = appsv1.StatefulSetSpec{
+			VolumeClaimTemplates: []v1.PersistentVolumeClaim{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: c.values.VolumeClaimTemplateName,
+					},
+					Spec: v1.PersistentVolumeClaimSpec{
+						AccessModes: []v1.PersistentVolumeAccessMode{
+							v1.ReadWriteOnce,
+						},
+						StorageClassName: c.values.StorageClass,
+						Resources:        getStorageReq(c.values),
+					},
+				},
+			},
+			PodManagementPolicy: appsv1.ParallelPodManagement,
+			UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+				Type: appsv1.RollingUpdateStatefulSetStrategyType,
+			},
+			Replicas:    pointer.Int32(c.values.Replicas),
+			ServiceName: c.values.ServiceName,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name":     "etcd",
+					"instance": c.values.EtcdName,
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: c.values.Annotations,
+					Labels:      c.values.Labels,
+				},
+				Spec: v1.PodSpec{
+					HostAliases: []v1.HostAlias{
+						{
+							IP:        "127.0.0.1",
+							Hostnames: []string{c.values.EtcdName + "-local"},
+						},
+					},
+					ServiceAccountName:        c.values.ServiceAccountName,
+					Affinity:                  c.values.Affinity,
+					TopologySpreadConstraints: c.values.TopologySpreadConstraints,
+					Containers: []v1.Container{
+						{
+							Name:            "etcd",
+							Image:           c.values.EtcdImage,
+							ImagePullPolicy: v1.PullIfNotPresent,
+							Command:         c.values.EtcdCommand,
+							ReadinessProbe: &v1.Probe{
+								Handler: v1.Handler{
+									Exec: &v1.ExecAction{
+										Command: c.values.ReadinessProbeCommand,
+									},
+								},
+								InitialDelaySeconds: 15,
+								PeriodSeconds:       5,
+								FailureThreshold:    5,
+							},
+							LivenessProbe: &v1.Probe{
+								Handler: v1.Handler{
+									Exec: &v1.ExecAction{
+										Command: c.values.LivenessProbCommand,
+									},
+								},
+								InitialDelaySeconds: 15,
+								PeriodSeconds:       5,
+								FailureThreshold:    5,
+							},
+							Ports:     getEtcdPorts(c.values),
+							Resources: getEtcdResources(c.values),
+							Env: []v1.EnvVar{
+								{
+									Name:  "ENABLE_TLS",
+									Value: c.values.EnableClientTLS,
+								},
+								{
+									Name:  "BACKUP_ENDPOINT",
+									Value: strconv.FormatBool(c.values.BackupTLS != nil),
```

Comment:
`BACKUP_ENDPOINT` is the backup-restore server endpoint, which would look like `http(s)://......:8080`. Please also set the correct `http`/`https` value based on whether `backupTLS` is enabled or not.

Reply:
Done